### PR TITLE
A small followup PR for PR 19310 

### PR DESCRIPTION
**Pull request by YaqiWang**
Add more documentations and a new test to make sure we can use subdomain names instead of subdomain ids for `SubdomainExtraElementIDGenerator` mesh generator. Refer to #19297. Tag @zachmprince and @GiudGiud

### CODE REVIEW

File: modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md [hunk 1]
```diff
@@ -4,9 +4,14 @@
 
 ## Overview
 
-The `SubdomainExtraElementIDGenerator` assigns element IDs based on mesh subdomain IDs of an input mesh.
-If the element IDs do not exist in the input mesh, they will be added by this mesh generator.
+The `SubdomainExtraElementIDGenerator` assigns extra element IDs based on mesh subdomain IDs of an input mesh.
+If the extra element IDs do not exist in the input mesh, this mesh generator will add them.
+The [!param](/Mesh/SubdomainExtraElementIDGenerator/subdomains) parameter can accept both subdomain names and subdomain IDs.
+The [!param](/Mesh/SubdomainExtraElementIDGenerator/extra_element_ids) parameter is a two-dimensional vector separated by semicolon and then white space as normal MOOSE input syntax for two-dimensional vectors.
```
Comment (GiudGiud) [inline modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md:10]: I think the easier way to convey this is to add a ## Example input syntax section at the end of the docs before the parameters

File: modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md [hunk 2]
```diff
@@ -4,9 +4,14 @@
...
+Its first dimensitionality must be equal to the size of the [!param](/Mesh/SubdomainExtraElementIDGenerator/extra_element_id_names) parameter and the size of each sub-vector must be equal to the size of the [!param](/Mesh/SubdomainExtraElementIDGenerator/subdomains) parameter.
+
 Default element IDs can be provided through [!param](/Mesh/SubdomainExtraElementIDGenerator/default_extra_element_ids) for elements in subdomains that are not specified in the [!param](/Mesh/SubdomainExtraElementIDGenerator/subdomains) parameter.
+Because of this, [!param](/Mesh/SubdomainExtraElementIDGenerator/default_extra_element_ids) is active only when the [!param](/Mesh/SubdomainExtraElementIDGenerator/subdomains) parameter does not include all mesh subdomains.
```
Comment (GiudGiud) [inline modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md:14]: this is already implied

**Review by GiudGiud — COMMENTED** [review]
are we concerned names wont work? There s no code change and additional coverage by adding this. This is essentially testing the behavior of getSubdomainID( name_or_id (but here a name)) , which is already tested

File: modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md (hunk 1 above)
Comment (GiudGiud) [reply modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md:10]: but this text is welcome too

**Comment by YaqiWang** [comment]
> are we concerned names wont work? 

I thought of this. I guess this does not hurt and give us a peace of mind on this functionality. I suggest that we keep this test.

File: modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md (hunk 2 above)
Comment (YaqiWang) [reply modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md:14]: Yes. But this sentence makes things clearer to users. They do not need to do the speculation.

File: modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md (hunk 1 above)
Comment (YaqiWang) [reply modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md:10]: I will add a `!listing`  here.

Comment (YaqiWang) [reply modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md:10]: https://mooseframework.inl.gov/docs/PRs/19530/site/source/meshgenerators/SubdomainExtraElementIDGenerator.html

Comment (GiudGiud) [reply modules/reactor/doc/content/source/meshgenerators/SubdomainExtraElementIDGenerator.md:10]: looks good

**Review by GiudGiud — DISMISSED** [review]

**Comment by YaqiWang** [comment]
@GiudGiud Can we merge this?

**Review by loganharbour — CHANGES_REQUESTED** [review]
Can you just use a mix of IDs and names in the first test instead?

**Comment by YaqiWang** [comment]
> Can you just use a mix of IDs and names in the first test instead?

Sounds good.

**Review by loganharbour — APPROVED** [review]

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**